src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -499,8 +582,29 @@ record Element(Label target, Label next, Object caseLabel) {}
                                              "charValue",
                                              MethodTypeDesc.of(ConstantDescs.CD_char));
                             cb.labelBinding(compare);
+
                             cb.ldc(integerLabel);
                             cb.if_icmpne(next);
+                        } else if ((element.caseLabel() instanceof Long ||
+                                    element.caseLabel() instanceof Float ||
+                                    element.caseLabel() instanceof Double ||
+                                    element.caseLabel() instanceof Boolean)) {
+                            //TODO: should call equals on the constant, not on the selector, check
```

Comment:
Looking at the code, it seems like we're already calling equals on the constant? (I assume that's to avoid spurious NPEs?)